src/main/java/org/opentripplanner/routing/core/AStarRequestBuilder.java

```diff
@@ -0,0 +1,120 @@
+package org.opentripplanner.routing.core;
+
+import java.time.Instant;
+import org.opentripplanner.model.GenericLocation;
+import org.opentripplanner.routing.api.request.StreetMode;
+import org.opentripplanner.routing.api.request.preference.RoutingPreferences;
+import org.opentripplanner.routing.api.request.request.VehicleParkingRequest;
+import org.opentripplanner.routing.api.request.request.VehicleRentalRequest;
+
+public class AStarRequestBuilder {
+
+  private Instant startTime = Instant.now();
+  private StreetMode mode = StreetMode.WALK;
+  private RoutingPreferences preferences = new RoutingPreferences();
+  private boolean arriveBy = false;
+  private boolean wheelchair = false;
+  private VehicleParkingRequest parking = new VehicleParkingRequest();
+  private VehicleRentalRequest rental = new VehicleRentalRequest();
+  private GenericLocation from = null;
+  private GenericLocation to = null;
+
+  AStarRequestBuilder() {}
+
+  AStarRequestBuilder(AStarRequest original) {
+    this.startTime = original.startTime();
+    this.mode = original.mode();
+    this.preferences = original.preferences().clone();
+    this.arriveBy = original.arriveBy();
+    this.wheelchair = original.wheelchair();
+    this.parking = original.parking().clone();
+    this.rental = original.rental();
+    this.from = original.from();
+    this.to = original.to();
+  }
+
+  public AStarRequestBuilder setStartTime(Instant startTime) {
+    this.startTime = startTime;
+    return this;
+  }
+
+  public Instant startTime() {
+    return startTime;
+  }
+
+  public AStarRequestBuilder setMode(StreetMode mode) {
+    this.mode = mode;
+    return this;
+  }
+
+  public StreetMode mode() {
+    return mode;
+  }
```

Comment:
Most of the time we do not need the accessors, but keep them if you do. In the preferences I kept a reference to the original object, because almost every time I wanted the original values, not the value in the builder. Then I just added a `original()` method to the builder to give access to it.

Reply:
In the `AStarRequest` constructor should I use direct field access and make them package-private, or should I modify the constructor to take each of the individual fields as arguments, rather than the builder object?